packages/client/src/v2-events/cache.ts

```diff
@@ -34,23 +34,76 @@ export function getUnsignedFileUrl(path: FullDocumentPath): FullDocumentUrl {
 }
 
 /**
- * Sets file to **BROWSER** cache with given filename
- * @see CACHE_NAME
+ * Waits for the workbox-runtime-* cache key to be available before proceeding.
+ * - Tries to find the cache key for CACHE_NAME up to 3 times with 500ms delay.
+ * - If not found, uses a localStorage retry counter to decide on reloads.
+ * - Clears storage only after 2 failed reloads to avoid infinite loops.
+ * - If still missing, clears all the storage and do a full reload (which redirects the user to login page).
+ * - If found, returns the found cache key.
  */
-export async function cacheFile({ url, file }: { url: string; file: File }) {
-  const temporaryBlob = new Blob([file], { type: file.type })
-  const cacheKeys = await caches.keys()
+export async function waitForWorkboxCacheKey() {
+  const MAX_ATTEMPTS = 3
+  const RETRY_DELAY_MS = 500
+  const RELOAD_LIMIT = 2
+  let cacheKey: string | undefined
 
-  const cacheKey = cacheKeys.find((key) => key.startsWith(CACHE_NAME))
+  for (let i = 0; i < MAX_ATTEMPTS; i++) {
+    const cacheKeys = await caches.keys()
+    cacheKey = cacheKeys.find((key) => key.startsWith(CACHE_NAME))
+    if (cacheKey) {
+      break
+    }
+    await new Promise((resolve) => setTimeout(resolve, RETRY_DELAY_MS))
+  }
 
   if (!cacheKey) {
-    // eslint-disable-next-line no-console
-    console.error(
-      `Cache ${CACHE_NAME} not found. Is service worker running properly?`
+    const reloadCounter = parseInt(
+      localStorage.getItem('workboxCacheReload') || '0',
+      10
     )
-    return
+    localStorage.setItem('workboxCacheReload', (reloadCounter + 1).toString())
+
+    if (reloadCounter < RELOAD_LIMIT) {
+      // eslint-disable-next-line no-console
+      console.warn(
+        `Cache ${CACHE_NAME} not found. Reloading page (attempt ${reloadCounter + 1})`
+      )
+      location.reload()
+    } else {
+      // eslint-disable-next-line no-console
+      console.error(
+        `Cache ${CACHE_NAME} still not found after multiple reloads. Clearing storage and reloading.`
+      )
+      localStorage.clear()
```

Comment:
This might be a bit too aggressive. Can we only clear out the `workboxCacheReload` and force a logout? I don't know if we need to delete all caches either. It might cause unwanted behaviour long-term